test/e2e/jmx/jmx_test.go

```diff
@@ -0,0 +1,411 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/util/awsservice"
+)
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags()
+}
+
+const (
+	NAMESPACE_JVM_TOMCAT        = "JVM_TOMCAT_E2E"
+	NAMESPACE_KAFKA             = "KAFKA_E2E"
+	NAMESPACE_CONTAINERINSIGHTS = "ContainerInsights/Prometheus"
+)
+
+var testRegistry = map[string][]func(*testing.T){
+	"jvm_tomcat.json": {
+		testTomcatMetrics,
+		testTomcatSessions,
+	},
+	"kafka.json": {
+		testKafkaMetrics,
+	},
+	"containerinsights.json": {
+		testContainerInsightsMetrics,
+		testTomcatRejectedSessions,
+	},
+}
+
+var nodeNames []string
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if flag.Lookup("test.run").Value.String() == "NO_MATCH" {
+		os.Exit(0)
+	}
+	env := environment.GetEnvironmentMetaData()
+
+	if env.Region != "us-west-2" {
+		if err := awsservice.ConfigureAWSClients(env.Region); err != nil {
+			fmt.Printf("Failed to reconfigure AWS clients: %v\n", err)
+			os.Exit(1)
+		}
+		fmt.Printf("AWS clients reconfigured to use region: %s\n", env.Region)
+	} else {
+		fmt.Printf("Using default testing region: us-west-2\n")
+	}
+
+	fmt.Println("Starting Helm installation...")
+	if err := ApplyHelm(env); err != nil {
+		fmt.Printf("Failed to apply Helm: %v\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Println("Waiting for metrics to propagate...")
+	time.Sleep(5 * time.Minute)
+
+	os.Exit(m.Run())
```

Comment:
Do the helm resources ever get cleaned up?

Reply:
With the exception of Load Balancers, they all should be removed upon the deletion of the cluster, but I'll look into handling this more elegantly.

Reply:
Added cleanup logic.